src/libappimage/desktop_integration/integrator/Integrator.cpp

```diff
@@ -224,30 +271,41 @@ namespace appimage {
                             iconPath /= (iconSize + "x" + iconSize);
                         }
 
-                        iconPath /= "apps";
+                        iconPath /= iconGroup;
                         iconPath /= iconNameBuilder.str();
 
                         auto deployPath = generateDeployPath(iconPath);
                         icon.save(deployPath.string(), icon.format());
-                    } catch (const IconHandleError& er) {
-                        Logger::error(er.what());
-                        Logger::error("No icon was generated for: " + appImage.getPath());
+                    } catch (const IconHandleError& error) {
+                        Logger::error("Icon deploy failed " + iconGroup + "/" + iconName
+                                      + " from " + appImage.getPath() + " : " + error.what());
                     }
                 }
 
                 /**
-                 * Append vendor prefix and appImage id to the file names to identify the appImage that owns
+                 * Prepend vendor prefix and appImage id to the file names to identify the appImage that owns
                  * this file. Replace the default XDG_DATA_DIR by the one at <xdgDataHome>
                  *
                  * @param path resource path
-                 * @return path with a prefixed file name
+                 * @return <xdg data home>/<relative path>/<vendor prefix>_<appImageId>_<file name>.<extension>
```

Comment:
You shouldn't hardcode output paths in the docs should you? Do you think that's really helpful to users? This is a level of abstraction, we might change it at any time, people should not rely on this information, otherwise you'll have to add more functions if this will ever be changed.

Reply:
This is how a hardcoded path looks like `/home/user/.local/share/applications/appimagekit_23542355_myapp.desktop` nothing to do with the one on the documentation which helps to understand what exactly is being prefixed to the file name. 

The `Integrator` class is a private class, the only components that will rely on it will be the libappimage components so it's safe to change.